utilities/pipelines/staticValidation/module.tests.ps1

```diff
@@ -802,6 +802,38 @@ Describe 'Test file tests' -Tag 'TestTemplate' {
             }
         }
     }
+    Context 'Role Assignmnet test' {
+
+        $roleAssignmentTestCases = @()
+
+        foreach ($moduleFolderPath in $moduleFolderPaths) {
+           $rbacFilePath = Join-Path $moduleFolderPath '.bicep/nested_roleAssignments.bicep'
+            if (Test-Path $rbacFilePath) {
+                $nestedFileContent = Get-Content $rbacFilePath 
+
+                #$resourceTypeIdentifier = $moduleFolderPath.Replace('\', '/').Split('/modules/')[1]
+                $resourceTypeIdentifier = (Get-Content (Join-Path $moduleFolderPath 'readme.md')).split('`')[1].substring(1).Replace(']', '')
+
+            }
+
+            $roleAssignmentTestCases += @{
+                resourceTypeIdentifier = $resourceTypeIdentifier
+                nestedFileContent      = $nestedFileContent
+                nestedFilePath         = $nestedFilePath
+            }
+        }
+
+        It '[<moduleFolderName>] nested role assignment deployment should contain the module resource Type as existing Resource.' -TestCases ($roleAssignmentTestCases | Where-Object { (Split-Path $_.nestedFilePath -Extension) -eq '.bicep' }) {
+
+            param(
+                [object[]] $nestedFileContent,
+                [string] $resourceTypeIdentifier
+            )
+
+            $hasExpectedResourceType = ($nestedFileContent | Out-String) -match $resourceTypeIdentifier
```

Comment:
While fine by itself, I would suggest to rather have a test that compares the used resource type with the one we expect as the `$resourceTypeIdentifier`. So the test would be like `$usedExistingResourceType | Should -Be $resourceTypeIdentifier`, which has the nice benefit, that a failing tests also tells you how you can fix the error.

To make this work however, you'd need to find the used resource type in the `$nestedFileContent`. This should be possible by searching for the line that matches `'resource ... existing = {'` and then extracting the used resource type.

This final step can be done using the snippet
```PowerShell
[regex]::Match($nestedFileContent[$existingResourceTypeRow], "^resource.+'([a-zA-Z\.\/]+)@.+' existing = {$").Captures.Groups[1].Value
```